osu.Game/Screens/Edit/Editor.cs

```diff
@@ -512,7 +512,21 @@ private void resetTrack(bool seekToStart = false)
 
         private void onModeChanged(ValueChangedEvent<EditorScreenMode> e)
         {
-            currentScreen?.Exit();
+            var lastScreen = currentScreen;
+
+            lastScreen?
+                .ScaleTo(0.98f, 200, Easing.OutQuint)
+                .FadeOut(200, Easing.OutQuint);
+
+            if ((currentScreen = screenContainer.FirstOrDefault(s => s.Type == e.NewValue)) != null)
```

Comment:
It feels like I'm beating a dead horse at this point, but I'd also use `SingleOrDefault` here. I realise it has performance implications but the container should have 4 children at absolute most and I'd really feel safer if this did check for duplicates, just in case.